source/extensions/filters/http/rate_limit_quota/client.h

```diff
@@ -25,13 +30,19 @@ class RateLimitQuotaCallbacks {
 };
 
 /**
- * A client used to query a rate limit quota service (RLQS).
+ * A client used to query the rate limit quota service (RLQS).
  */
 class RateLimitClient {
 public:
   virtual ~RateLimitClient() = default;
 
-  virtual void rateLimit(RateLimitQuotaCallbacks& callbacks) PURE;
+  virtual absl::Status startStream(const StreamInfo::StreamInfo& stream_info) PURE;
+  virtual void closeStream() PURE;
+  virtual void
+  sendUsageReport(absl::optional<envoy::service::rate_limit_quota::v3::BucketId> bucket_id) PURE;
```

Comment:
BucketId is a proto. It will be copied every time this method is called. It makes sense to use a reference here.